aea/aea.py

```diff
@@ -160,33 +167,25 @@ def handle(self, envelope: Envelope) -> None:
             error_handler.send_invalid_message(envelope)    # pragma: no cover
             return                                          # pragma: no cover
 
-        handlers = self.resources.handler_registry.fetch(protocol.id)
+        handlers = self.filter.get_active_handlers(protocol.id)
         if handlers is None:
             if error_handler is not None:
                 error_handler.send_unsupported_skill(envelope)
```

Comment:
should it really be an "unsupported skill" error? Should discern between "inactive" handler and "unsupported" handler?

Reply:
I am not sure. Basically, this is just telling the other agent: "I cannot deal with this". We probably should keep private to the agent the info why that is (i.e. unsupported or not active skill). So the point I would make is that it should maybe be renamed.